PyPDF2/_utils.py

```diff
@@ -413,3 +414,25 @@ def rename_kwargs(  # type: ignore
                     f"{old_term} is deprecated as an argument. Use {new_term} instead"
                 )
             )
+
+
+@dataclass
+class File:
+    name: str
+    data: bytes
+    mime_type: str
+
+    @property
+    def file_extension(self) -> str:
```

Comment:
We should follow pillow's lead here and call this function `format` to match theirs, as that's technically what this is, not the "extension" as one would expect for a "file". There's nothing to say the extension of a file needs to match the mime type / format, e.g. if I do `File(name='foo.txt', mime_type='image/png')`, the extension is `.txt`, but the format is `png`.

```suggestion
    def format(self) -> str:
```